auth/forms.py

```diff
@@ -4,13 +4,25 @@
 
 from backend.models import UserCourse, Course
 from django.contrib.auth.models import Group
+from django.conf import settings
 
 
 class SignupForm(forms.Form):
 
+    def clean(self):
+        super(SignupForm, self).clean()
+        course_codes = self.data.get('course_code').replace(' ', '').split(';')
+        for course_code in course_codes:
+            #if the course code is empty or is neither an instructor nor a valid student course code
+            if not course_code or (course_code != settings.INSTRUCTOR_CODE and
+                                           Course.objects.filter(code=course_code).count() < 1):
+                raise forms.ValidationError('COURSE CODE is invalid', code="invalid")
+            #Later we could have the admin assign a unique code to an instructor for registering
```

Comment:
General nit, we prefer to line wrap at 80.  You can run pep8 against this to check this sort of thing.